### PR TITLE
Lab1

### LINKED ISSUE #338: Lab 1 - Fibonacci
Create an executable go program in directory 01_fib/USERNAME
Write a function (func fib(n int)) that prints the first n Fibonacci numbers
For negative n print Negafibonacci (func negaFib(n int)) numbers.

### PR DESCRIPTION
**Pull request by k-vikingsson**
Fixes #338 

Review of colleague's PR #

#### Changes proposed in this PR:

 - Implementation of fib() and negaFib()

### CODE REVIEW

**Review by undrewb — APPROVED** [review]
Negafib is a special case of fib. Consider refactoring to use less code.

In test you don't have tests for each function. In a simple example this doesnt matter so much but it potentially makes testing harder. Consider having tests for all functions.

Approving because it works and the logic is there.

File: 01_fib/k-vikingsson/main_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"testing"
+)
+
+var buf bytes.Buffer
+
+var testCases = map[string]struct {
+	input    int
+	expected string
+}{
+	"Basic":    {input: 7, expected: "1\n1\n2\n3\n5\n8\n13\n"},
```
Comment (juliaogris) [inline 01_fib/k-vikingsson/main_test.go:14]: Assignment asked for output starting at `1` not `0`

Comment (juliaogris) [inline 01_fib/k-vikingsson/main_test.go:14]: Assignment asked for output starting at `1` not `0`

File: 01_fib/k-vikingsson/main.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		negaFib(n)
+		return
+	}
+	c := 1
+	nA := 1
+	nB := 1
+	nC := nA + nB
+	for c <= n {
+		switch c {
+		case 1:
```
Comment (juliaogris) [inline 01_fib/k-vikingsson/main.go:43]: Pull out of loop.

**Review by juliaogris — CHANGES_REQUESTED** [review]
Please [fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages) with `git rebase -i HASH` and read through the [Collected Code Review Comments](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments).

Adding a `switch` or `if` whose conditionals depend of the value of the loop index can be improved by pulling the special cases out of the loop so that no  tests are executed on every iteration.

Whilst your solution works, it's a bit more verbose than could be. Here our reference implementation, feel free to take it, adjust it or ignore it:
```go
func fib(n int) {
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		fmt.Fprintln(out, n1)
	}
}
```

**Pull request closed without merging**